### PR TITLE
feat: Handling invalid RDF in canonized JSON LD

### LINKED ISSUE #1592: Signinig & Verification: Incorrect canonicalization of credential status
During interop tests noticed an issues with canonicalization of verifiable credentials with credentials status.

Sample Doc : 

```
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://www.w3.org/2018/credentials/examples/v1"
  ],
  "credentialSchema": [],
  "credentialStatus": {
    "id": "http://issuer.vc.rest.example.com:8070/status/1",
    "type": "CredentialStatusList2017"
  },
  "credentialSubject": {
    "degree": {
      "degree": "MIT",
      "type": "BachelorDegree"
    },
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "name": "Jayden Doe",
    "spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1"
  },
  "id": "https://example.com/credentials/932236e0-966c-44cf-9342-236c0a2c77a7",
  "issuanceDate": "2020-03-16T22:37:26.544Z",
  "issuer": {
    "id": "did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg",
    "name": "alice_f94db66c-be63-4f03-af10-4205d1f625e1"
  },
  "proof": {
    "created": "2020-04-08T04:00:22Z",
    "jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..6ICruO8hE3mYEyWeM01aB9iPmsGqhZg8rpNXPTofUUulYXJUJ2DOLe0uv1KJ3bR4N71h_Hp_S3YmzIUpyeoNCw",
    "proofPurpose": "assertionMethod",
    "type": "Ed25519Signature2018",
    "verificationMethod": "did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg#xqc3gS1gz1vch7R3RvNebWMjLvBOY-n_14feCYRPsUo"
  },
  "type": [
    "VerifiableCredential",
    "UniversityDegreeCredential"
  ]
}
```

**After canonicalization using JSON LD LIB** (https://github.com/digitalbazaar/jsonld-signatures/tree/master/lib)

```
<did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg> <http://schema.org/name> "alice_6edf241c-ac8e-4c31-8c53-a0b0e69d937e"^^<http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML> .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <http://schema.org/name> "Jayden Doe"^^<http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML> .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <http://schema.org/spouse> "did:example:c276e12ec21ebfeb1f712ebc6f1" .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <https://example.org/examples#degree> _:c14n0 .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://example.org/examples#UniversityDegreeCredential> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/2018/credentials#VerifiableCredential> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#credentialStatus> <http://issuer.vc.rest.example.com:8070/status/1> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#credentialSubject> <did:example:ebfeb1f712ebc6f1c276e12ec21> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#issuanceDate> "2020-03-16T22:37:26.544Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#issuer> <did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg> .
_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://example.org/examples#BachelorDegree> .
_:c14n0 <https://example.org/examples#degree> "MIT" .
```

**Using Aries Framework Go:**

```
<did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg> <http://schema.org/name> "alice_6edf241c-ac8e-4c31-8c53-a0b0e69d937e"^^<http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML> .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <http://schema.org/name> "Jayden Doe"^^<http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML> .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <http://schema.org/spouse> "did:example:c276e12ec21ebfeb1f712ebc6f1" .
<did:example:ebfeb1f712ebc6f1c276e12ec21> <https://example.org/examples#degree> _:c14n0 .
<http://issuer.vc.rest.example.com:8070/status/1> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <CredentialStatusList2017> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://example.org/examples#UniversityDegreeCredential> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/2018/credentials#VerifiableCredential> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#credentialStatus> <http://issuer.vc.rest.example.com:8070/status/1> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#credentialSubject> <did:example:ebfeb1f712ebc6f1c276e12ec21> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#issuanceDate> "2020-03-16T22:37:26.544Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<https://example.com/credentials/bea1bc1e-054d-4ceb-b72c-f2f757e93b5d> <https://www.w3.org/2018/credentials#issuer> <did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg> .
_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://example.org/examples#BachelorDegree> .
_:c14n0 <https://example.org/examples#degree> "MIT" .
```

### PR DESCRIPTION
**Pull request by sudeshrshetty**
Previously: we weren't dropping undefined terms from RDFs.

Fix: changed normalization aproach through json-gold library so that it
returns parsing error whenever invalid data found in dataset.

Added error handling logic for invalid RDF data errors where aries json
ld processor is going to remove the invalid data from dataset and try
again recursively. (Following
https://github.com/digitalbazaar/jsonld.js/issues/199)

closes #1592

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: pkg/doc/signature/jsonld/processor.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	"strings"
+
+	"github.com/piprate/json-gold/ld"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+const (
+	format             = "application/n-quads"
+	algorithm          = "URDNA2015"
```
Comment (troyronda) [inline pkg/doc/signature/jsonld/processor.go:22]: should  any of these be options?

File: pkg/doc/signature/suite/ecdsasecp256k1signature2019/suite.go [hunk 2]
```diff
@@ -14,26 +14,24 @@ package ecdsasecp256k1signature2019
...
 type Suite struct {
 	suite.SignatureSuite
+	jsonldProcessor *jsonld.Processor
 }
 
 const (
 	signatureType = "EcdsaSecp256k1Signature2019"
 	jwkType       = "EcdsaSecp256k1VerificationKey2019"
-	format        = "application/n-quads"
-	algorithm     = "URDNA2015"
```
Comment (troyronda) [inline pkg/doc/signature/suite/ecdsasecp256k1signature2019/suite.go:31]: i think the algorithm constant should be with the suite

Comment (troyronda) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/suite.go:31]: as i see it defined in the suite. eg, https://w3c-ccg.github.io/lds-ed25519-2018

File: pkg/doc/signature/jsonld/processor.go (hunk 1 above)
Comment (sudeshrshetty) [reply pkg/doc/signature/jsonld/processor.go:22]: will fix this, but right now json-gold only support this algorithm. Not sure if this algorithm is something to do with suite.

[Universal RDF Dataset Normalization Algorithm 2015 or URDNA2015.](http://json-ld.github.io/normalization/spec/)

File: pkg/doc/signature/suite/ecdsasecp256k1signature2019/suite.go (hunk 2 above)
Comment (sudeshrshetty) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/suite.go:31]: will fix it

File: pkg/doc/signature/jsonld/processor.go (hunk 1 above)
Comment (troyronda) [reply pkg/doc/signature/jsonld/processor.go:22]: if your below logic is tied to URDNA - eg retryForInvalidRDF then i guess it couldn’t be a parameter.

Comment (sudeshrshetty) [reply pkg/doc/signature/jsonld/processor.go]: done

**Pull request merged**